### PR TITLE
fix: syncfile missing

### PR DESCRIPTION
**Pull request by okdistribute**
a syncfile with missing data will hang the replication

### CODE REVIEW

File: test/sync.js [hunk 1]
```diff
@@ -1133,6 +1133,113 @@ tape('sync: missing data still ends', function (t) {
   })
 })
 
+tape.only('sync: missing data syncfile still ends', function (t) {
```
Comment (hackergrrl) [inline test/sync.js:1136]: Reminder that this is still `tape.only` :3

File: test/sync.js [hunk 2]
```diff
@@ -1133,6 +1133,113 @@ tape('sync: missing data still ends', function (t) {
...
+    function restart () {
+      api1.sync.leave()
+      api1.close((err) => {
+        api1.osm.close(() => {
+          t.error(err, 'closed first api')
+          _api1 = helpers.createApi(api1._dir)
+          helpers.writeBigDataNoPhotos(_api1, 200, () => {
+            _api1.sync.listen(() => {
+              api2.sync.once('peer', (peer) => {
+                sync(peer)
```
Comment (gmaclennan) [inline test/sync.js:1178]: Missing callback arg

File: test/sync.js [hunk 3]
```diff
@@ -1133,6 +1133,113 @@ tape('sync: missing data still ends', function (t) {
...
+    helpers.writeBigData(api1, 50, written)
+    helpers.writeBigDataNoPhotos(api2, 450, written)
+
+    function written (err) {
+      t.error(err, 'written no error')
+      if (--pending === 0) {
+        t.ok(api1.sync.peers().length > 0, 'api 1 has peers')
+        t.ok(api2.sync.peers().length > 0, 'api 2 has peers')
+        if (api2.sync.peers().length >= 1) {
+          sync(api2.sync.peers()[0])
```
Comment (gmaclennan) [inline test/sync.js:1164]: missing callback arg

File: test/sync.js [hunk 4]
```diff
@@ -1133,6 +1133,113 @@ tape('sync: missing data still ends', function (t) {
...
+    }
+
+    function sync (peer, cb) {
+      if (!cb) cb = () => {}
+      t.ok(peer, 'syncronizing ' + numSyncs)
+      numSyncs++
+      api2.sync.once('down', (downPeer) => {
+        t.pass('emit down event on close')
+        t.notOk(downPeer.connected, 'not connected anymore')
+        if (numSyncs === 2) {
```
Comment (gmaclennan) [inline test/sync.js:1215]: I think this just hangs here if `numSyncs !== 2` because callback is noop?

**Review by gmaclennan — COMMENTED** [review]
@okdistribute just coming across this during some maintenance / cleanup. If you have a moment, I'm not sure about the intended logic in the test, but it seems to be hanging, could you take a quick look?

**Pull request closed without merging**